### DISCUSSION TITLE
LanguageAgnosticInlining deprecation

### DISCUSSION THREAD

**Discussion by petenorth** [oracle/truffleruby#2134] [General] [answered]
Do the TruffleRuby team run any benchmarks with engine.LanguageAgnosticInlining=true/false ? The flag is being deprecated so presumably it will always be switched on in forthcoming releases. On short lived processes (i.e. CI type things) running with engine.LanguageAgnosticInlining=false is faster when in throughput mode but the flag is only really relevant for long lived processes hence the question.

**Comment by eregon** [comment, marked as answer]
The old Truffle inlining (`--engine.LanguageAgnosticInlining=false`) is essentially deprecated at this point, which is why the option is deprecated.
No, we don't run benchmarks with that option anymore (we did to compare for a while).

How much faster is it in your case?
I guess MultiTier (should reach same peak) or Mode=latency (much slower peak but faster warmup than default) is more important for CI things.

**Reply by petenorth** [reply]
I've looked again at different runs and I think what I see is that in latency mode there isn't much difference between running with language agnostic inlining or not.

When running in throughput mode there is a consistent marginal slow down with rspec. But like I say it doesn't make sense to run rspec in throughput mode anyway.

I think I'll wait until multitier compilation becomes the default and then compare (i.e. relying only on defaults )

**Reply by eregon** [reply]
MultiTier is default in head builds and will be in the next release.
So I think the defaults should be fairly good now, Mode=latency might still be faster for rspec.
It would be interesting to compare again and post the results here or in https://github.com/oracle/truffleruby/issues/1985

**Reply by petenorth** [reply]
Posted an update on #1985

**Reply by eregon** [reply]
Thanks!
